src/SSHDebugPS/UI/ContainerPickerDialogWindow.xaml

```diff
@@ -0,0 +1,309 @@
+                IsEnabled="{Binding CanAddConnection}">
+            <vsimaging:CrispImage Moniker="{x:Static catalog:KnownMonikers.Add}"
+                                  Width="16"
+                                  Height="16">
+                <Image.ToolTip>
+                    <TextBlock Text="Add New Connection" />
+                </Image.ToolTip>
+            </vsimaging:CrispImage>
+        </Button>
+
+        <!-- Status Text and Refresh Button -->
+
+        <TextBlock x:Uid="uidStatusLabel"
+                    Style="{StaticResource StatusLabelStyle}"
+                    Grid.Row="2"
+                    Grid.Column="0"
+                    Margin="0,10,0,10"
+                    Padding="10,0,10,0"
+                    Text="{ Binding StatusText, Mode=OneWay}"
+                    TextWrapping="Wrap"/>
+
+        <Button x:Uid="uidRefreshButton"
+                    Grid.Row="2"
+                    Grid.Column="1"
+                    AutomationProperties.Name="Refresh"
+                    Margin="0,10,0,10"
+                    HorizontalAlignment="Right"
+                    VerticalAlignment="Center"
+                    Height="20"
+                    Width="20"
+                    Click="Refresh_Click"
+                    IsEnabled="{Binding IsRefreshEnabled}">
+            <vsimaging:CrispImage Moniker="{x:Static catalog:KnownMonikers.Refresh}"
+                                  Width="16"
+                                  Height="16">
+                <Image.ToolTip>
+                    <TextBlock Text="Refresh" />
+                </Image.ToolTip>
+            </vsimaging:CrispImage>
+        </Button>
+
+        <!-- Container Instance display-->
+        <ScrollViewer AutomationProperties.Name="ConnectionScrollViewer"
+                      Style="{StaticResource ScrollViewerStyle}"
+                      Grid.Row="4"
+                      Grid.ColumnSpan="2">
+            <Grid Grid.Row="1">
+                <Grid.RowDefinitions>
+                    <RowDefinition Height="*" />
+                </Grid.RowDefinitions>
+
+                <ListBox x:Name="ContainerListBox"
+                         AutomationProperties.AutomationId="aidContainerListBox"
+                         ItemsSource="{Binding Path=ContainerInstances, Mode=OneWay }"
+                         SelectedValue="{Binding SelectedContainerInstance}"
+                         ItemContainerStyle="{StaticResource ListBoxContainerStyle}"
+                         GotKeyboardFocus="ContainerListBox_GotKeyboardFocus"
+                         Style="{StaticResource ListBoxStyle}"/>
+            </Grid>
+        </ScrollViewer>
+
+        <!-- OK / Cancel buttons -->
+        <StackPanel Grid.Row="5"
+                    Grid.ColumnSpan="2"
+                    Margin="4"
+                    Orientation="Horizontal"
+                    HorizontalAlignment="Right">
+
+            <Button x:Uid="uidOkayButton" x:Name="OkayButton"
```

Comment:
>Button [](start = 13, length = 6)

Do we need to disable the okay button if no containers are selected?

Reply:
currently we treat it like a cancel where nothing is selected so we return no data. I can talk to Leslie and see if that is acceptable or if we should just grey out "ok". Doing it would probably be trivial.

---
In reply to: [287538546](https://github.com/microsoft/MIEngine/pull/868#discussion_r287538546) [](ancestors = 287538546)